samples/manageHostname.py

```diff
@@ -0,0 +1,43 @@
+import argparse
+import vm_automation
+
+def updateHostname(vm, hostname):
+    # Remember machine state and ensure powered on
+    powerState = self.isPoweredOn()
+    if powerState == False:
+        self.powerOn
+
+    # Windows path
+    if ("windows" in self.vmOS):
+        self.runCmdOnGuest("wmic computersystem where name=\"%COMPUTERNAME%\" rename \"" + hostname + "\"")
+    # Linux path
+    else:
+        self.runCmdOnGuest("sudo sed \'s/$HOSTNAME/" + hostname + "/\' /etc/hosts | sudo sed \'s/$HOSTNAME/" + hostname + "/\' /etc/hostname")
+
+    # Return machine to original state
+    if powerState == False:
+        self.powerOff
```

Comment:
Hostname changes usually require a reboot to fully apply, consider ensuring a reboot and possibly a validation step to ensure the change applied before returning to the original state.

This may required some sort of wait or check to ensure the host is ready after reboot.

Also consider using safe power controls if tools are installed.
https://github.com/rapid7/vm-automation/blob/2fc6268c95e22b139878e70a88e8592d1ab22b18/samples/managePower.py#L21-L22